api.py

```diff
@@ -96,16 +105,19 @@ async def validate_schema(data):
     return response
 
 
-def is_hostname_allowed(hostname):
-    allowed_full_domains = {
-        "gist.githubusercontent.com",
-        "raw.githubusercontent.com",
-    }
-    allowed_base_domains = {"onsdigital.uk"}
-
-    return hostname in allowed_full_domains or any(
-        hostname.endswith(f".{base}") for base in allowed_base_domains
+def is_domain_allowed(parsed_url):
+    base_url = f"{parsed_url.scheme}://{parsed_url.netloc}/"
+    repo_owner = (
+        parsed_url.path.split("/")[1] if len(parsed_url.path.split("/")) > 1 else ""
     )
+    domain = parsed_url.netloc
```

Comment:
Should be created only once above and called twice.

Reply:
Updated 👍